SendFire.Common/Process/CommandLineExecutionProvider.cs

```diff
@@ -0,0 +1,274 @@
+﻿using SendFire.Common.ExtensionMethods;
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Collections.Generic;
+
+namespace SendFire.Common.Process
+{
+    public class CommandLineExecutionProvider
+    {
+
+        public List<CommandExecutionParamModel> _commandLineModels;
+        private bool _runAsBatch;
+        //public string ProcessOutput { get; set; }
+        public CommandLineExecutionProvider(string[] args, bool runAsBatch) {
+            if (args.Length < 1)
+            {
+                throw new ArgumentException("No command line specified.");
+            }
+            _runAsBatch = runAsBatch;
+            _commandLineModels = ParseArguments(args);
+        }
+
+        private List<CommandExecutionParamModel> ParseArguments(string[] args)
```

Comment:
commands instead of args, change this IN ALL PLACES it is a command list and not an argument list